crates/rpc/rpc/src/eth/cache.rs

```diff
@@ -380,6 +380,41 @@ where
     }
 }
 
+impl<Client, Tasks> EthStateCacheService<Client, Tasks>
+where
+    Client: StateProviderFactory + BlockProvider + EvmEnvProvider + Clone + Unpin + 'static,
+    Tasks: TaskSpawner + Clone + 'static,
+{
+    /// Check for new blocks and insert them in cache
+    async fn check_for_new_blocks<St>(&mut self, mut events: St)
+    where
+        St: Stream<Item = CanonStateNotification> + Unpin + 'static,
+    {
+        while let Some(event) = events.next().await {
+            match event {
+                CanonStateNotification::Commit { new } => {
+                    let (new_blocks, _) = new.inner();
+                    self.insert_new_block_into_cache(new_blocks.tip().clone().hash).await;
+                }
+                CanonStateNotification::Reorg { old: _, new } => {
+                    let (new_blocks, _) = new.inner();
+                    self.insert_new_block_into_cache(new_blocks.tip().clone().hash).await;
+                }
+                _ => {}
+            }
+        }
+    }
+
+    async fn insert_new_block_into_cache(&mut self, block_hash: H256) {
+        let client = self.client.clone();
+        let action_tx = self.action_tx.clone();
+        self.action_task_spawner.spawn(Box::pin(async move {
+            let res = client.block_by_hash(block_hash);
+            let _ = action_tx.send(CacheAction::BlockResult { block_hash, res });
+        }));
+    }
```

Comment:
this is now essentially redundant because this only instructs the cache to lookup blocks from disk